piccolo_api/media/azure.py

```diff
@@ -0,0 +1,432 @@
+from __future__ import annotations
+
+import asyncio
+import datetime
+import functools
+import io
+import os
+import pathlib
+import sys
+import typing as t
+from concurrent.futures import ThreadPoolExecutor
+
+from piccolo.apps.user.tables import BaseUser
+from piccolo.columns.column_types import Array, Text, Varchar
+
+from .base import ALLOWED_CHARACTERS, ALLOWED_EXTENSIONS, MediaStorage
+from .content_type import CONTENT_TYPE
+
+if t.TYPE_CHECKING:  # pragma: no cover
+    from concurrent.futures._base import Executor
+
+try:
+    from azure.identity import DefaultAzureCredential
+    from azure.storage.blob import (
+        BlobSasPermissions,
+        BlobServiceClient,
+        ContainerClient,
+        generate_blob_sas,
+    )
+
+    azure_import_successful = True
+except ImportError:  # pragma: no cover
+    azure_import_successful = False
+
+
+class AzureMediaStorage(MediaStorage):
+    def __init__(
+        self,
+        column: t.Union[Text, Varchar, Array],
+        storage_account_name: str,
+        container_name: str,
+        folder_name: t.Optional[str] = None,
+        connection_kwargs: t.Optional[t.Dict[str, t.Any]] = None,
```

Comment:
We don't really use `connection_kwargs` for much at the moment - it is just being used as a way of passing a connection string.

Should we pass the `connection_kwargs` to `DefaultAzureCredential`? I know the user can set most of the options via environment variables, but it might be convenient to pass them in directly.

https://learn.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python#constructor

In which case, connection string, could be a separate argument, or:

```python
connection_kwargs: t.Optional[t.Dict[str, t.Any]] | str = None
```

If it's a string, we treat it as a connection string.

What do you think?

Reply:
I thought about it, I think instead of having a union of an optional dictionary or string, leave it as an optional dictionary and if `'connection_string'` is not present, pass the kwargs to DefaultCredential